### ISSUE TITLE
'XDG_RUNTIME_DIR not set' error in deCONZ version 2.05.84

### ISSUE THREAD

**Issue by corneyl** [marthoc/docker-deconz#297]
I get the following error when updating from 2.05.79 to the newest stable version of deCONZ:
```
[marthoc/deconz] Starting deCONZ...
[marthoc/deconz] Current deCONZ version: 2.05.84
[marthoc/deconz] Web UI port: 8080
[marthoc/deconz] Websockets port: 8443
[marthoc/deconz] VNC Disabled
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
QStandardPaths: wrong ownership on runtime directory /tmp/runtime-root, -2 instead of 0
libpng warning: iCCP: known incorrect sRGB profile
This plugin does not support propagateSizeHints()
This plugin does not support propagateSizeHints()
This plugin does not support propagateSizeHints()
```

When I set a value for `XDG_RUNTIME_DIR`, e.g. `XDG_RUNTIME_DIR=/root/xdg`, I get the following:
```
[marthoc/deconz] Starting deCONZ...
[marthoc/deconz] Current deCONZ version: 2.05.84
[marthoc/deconz] Web UI port: 8080
[marthoc/deconz] Websockets port: 8443
[marthoc/deconz] VNC Disabled
QCoreApplication::arguments: Please instantiate the QApplication object first
QStandardPaths: XDG_RUNTIME_DIR points to '/root/xdg' which is not a directory
libpng warning: iCCP: known incorrect sRGB profile
This plugin does not support propagateSizeHints()
This plugin does not support propagateSizeHints()
This plugin does not support propagateSizeHints()
```

Container gets started using `docker-compose`:
```
deconz:
  image: marthoc/deconz:armhf-2.05.84
  ports:
    - 8080:8080
  networks:
    - ha-net
  restart: always
  volumes:
    - ./data/deconz:/root/.local/share/dresden-elektronik/deCONZ
    - ./data/zigbee-otau-fw:/root/otau
  devices:
    - /dev/ttyAMA0
  environment:
    - DECONZ_WEB_PORT=8080
    - DECONZ_WS_PORT=8443
    - DEBUG_INFO=1
    - DEBUG_APS=0
    - DEBUG_ZCL=0
    - DEBUG_ZDP=0
    - DEBUG_OTAU=1
```

It seems related to Qt and is probably related to #295, can you confirm this @Kane610?

**Comment by marthoc** [comment]
I think this is related to the switch to Debian. The solution is probably for me to create a directory and set that variable in the Dockerfile. But what is more interesting to me is why on your system the permission is not being set correctly (on mine I also get that error, but the default directory gets created OK, so it doesn't fail).

**Comment by corneyl** [comment]
I did look in the container to see if the directory was created, and is was in both situations. So `/tmp/runtime-root` or `/root/xdg` got created with `root:root` as owner. But the application doesn't start any further than shown in the logs above. 

So maybe there's another problem going on that's not related to the error messages?

**Comment by marthoc** [comment]
Possibly, I am looking into a fix.

**Comment by marthoc** [comment]
Can you test the image marthoc/deconz:amd64-test-ubuntu for me? I switched the base to ubuntu just to see if this error still occurs.

**Comment by corneyl** [comment]
Unfortunately the test image has the wrong architecture, I'm running Deconz on a Pi so need an image in the armhf/armv7 architecture.

**Comment by marthoc** [comment]
@corneyl Something else to try is running the container with `—privileged`... it appears that on some host systems/kernels this may be necessary, I am still researching why.

**Comment by Rett1** [comment]
> @corneyl Something else to try is running the container with `—privileged`... it appears that on some host systems/kernels this may be necessary, I am still researching why.

Running the latest image in privileged mode solves the `wrong ownership` issue as mentioned in [#295](https://github.com/marthoc/docker-deconz/issues/295)
XDG_RUNTIME_DIR message remains.


> Can you test the image marthoc/deconz:amd64-test-ubuntu for me? I switched the base to ubuntu just to see if this error still occurs.

Just tried it (without privileged). deConz and Phoscon are working fine. The `wrong ownership` message does not appear in the log anymore, but `QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'` remains.

**Comment by corneyl** [comment]
I also just tried the `armhf-2.05.84` image in privileged mode and that one runs fine for me too.

**Comment by TorbjornZachrisson** [comment]
I had the same problems. Just switching to image marthoc/deconz:amd64-test-ubuntu worked instantly.

**Comment by marthoc** [comment]
Looks like a switch to a Ubuntu base might be in the works!

**Comment by Rett1** [comment]
Seems to be fixed in `2.05.86`. No related Error messages anymore. Thanks!

**Comment by Mimiix** [comment]
As there hasn't been any new replies and as there are newer versions, i'm closing this issue. If this needs to be re-opened, please ping me :)
